core/src/main.cpp

```diff
@@ -235,24 +235,41 @@ constexpr auto default_jsonschema() -> std::string_view
                                     "minimum": 1
                                 }},
                                 "provider_url": {{
-                                    "type": "string"
+                                    "type": "string",
+                                    "format": "uri"
+                                }},
+                                "run_as_protected_resource": {{
```

Comment:
discush... is this in addition to running as an oidc client?  or instead of?  do we want to create a 'mode' instead of a boolean for this?

Reply:
An enum definitely gives us more flexibility.

Reply:
Leave as is & create issue for this?

Reply:
Let's go ahead and handle this.

Post what you think the enum property should be called and what values it should accept.

Reply:
To clarify, do we expect an 'oidc_mode' and have it be 'protected_resource' or 'client' for example? Is this better than a boolean?

Reply:
Introducing a mode just gives us more flexibility if/when we need to change things. The boolean only gives us two values to work with.

Reply:
Gotcha. In that case, are we going to require the OIDC section, or have two ways to disable OIDC via 'none' and omission?

Reply:
after discussion...

- we'll use basic if it's defined
- we'll use openid_connect if it's defined

no conditional logic or 'mode' for now.

Reply:
and 'mode' for 'client' or 'protected_resource' within openid_connect.

Reply:
Done, leaving open just in case we prefer a different name than 'mode'.